rust/src/measurements/make_private_expr/ffi.rs

```diff
@@ -0,0 +1,39 @@
+use polars_plan::dsl::Expr;
+
+use crate::{
+    core::{FfiResult, IntoAnyMeasurementFfiResultExt},
+    domains::ExprDomain,
+    ffi::{
+        any::{AnyDomain, AnyMeasure, AnyMeasurement, AnyMetric, AnyObject, Downcast},
+        util,
+    },
+    measures::MaxDivergence,
+    metrics::SymmetricDistance,
+};
+
+use super::make_private_expr;
+
+#[no_mangle]
+pub extern "C" fn opendp_measurements__make_private_expr(
+    input_domain: *const AnyDomain,
+    input_metric: *const AnyMetric,
+    output_measure: *const AnyMeasure,
+    expr: *const AnyObject,
+    param: *const AnyObject,
+) -> FfiResult<*mut AnyMeasurement> {
+    let input_domain = try_!(try_as_ref!(input_domain).downcast_ref::<ExprDomain>()).clone();
+    let in_metric = try_!(try_as_ref!(input_metric).downcast_ref::<SymmetricDistance>()).clone();
+    let out_measure =
+        try_!(try_as_ref!(output_measure).downcast_ref::<MaxDivergence<f64>>()).clone();
+
+    let param = if let Some(param) = util::as_ref(param) {
+        try_!(try_as_ref!(param).downcast_ref::<f64>()).clone()
+    } else {
+        f64::NAN
+    };
+
+    let expr = try_!(try_as_ref!(expr).downcast_ref::<Expr>()).clone();
+    make_private_expr(input_domain, in_metric, out_measure, expr, param)
```

Comment:
Reuse names, like you do for `input_domain`? All of these seem to be cleaned-up versions of the originals, and it would be bad to accidentally refer to the originals. (If the meaning of a variable does change, the new name can reflect the new meaning.)

```suggestion
    let input_domain = try_!(try_as_ref!(input_domain).downcast_ref::<ExprDomain>()).clone();
    let input_metric = try_!(try_as_ref!(input_metric).downcast_ref::<SymmetricDistance>()).clone();
    let output_measure =
        try_!(try_as_ref!(output_measure).downcast_ref::<MaxDivergence<f64>>()).clone();

    let param = if let Some(param) = util::as_ref(param) {
        try_!(try_as_ref!(param).downcast_ref::<f64>()).clone()
    } else {
        f64::NAN
    };

    let expr = try_!(try_as_ref!(expr).downcast_ref::<Expr>()).clone();
    make_private_expr(input_domain, input_metric, output_measure, expr, param)
```